Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/LuceneIndexController.java

```diff
@@ -55,24 +58,26 @@ public String onLoadLucene(ModelMap model) {
      * @return
      * @throws IndexerRunningException
      */
+    @Async
     @RequestMapping(value = "auth/indexConcepts", method = RequestMethod.POST, produces = "application/json")
-    public @ResponseBody IndexingEvent indexConcepts(HttpServletRequest req, Principal principal, ModelMap model)
+    public @ResponseBody Future<IndexingEvent> indexConcepts(HttpServletRequest req, Principal principal,
```

Comment:
not sure how you are polling. where are you checking the response of this method for being done? In any case, you should start the new thread with your indexer not your controller. Doing it at the controller isn't a very clean approach since the long running task is the indexing not the controller method. You loose control over the execution that way.